R/Rscape/build_tarballs.jl

```diff
@@ -0,0 +1,155 @@
+
+# Note: license file for R-view is missing, it claims to be GPLv3 but
+#       doesn't contain a full copy of the license.
+echo \"""
+The source code for R-view doesn't include a license file, and the only note in the
+source code is in lib/R-view/src/rview_config.h which says it is GPLv3.
+This gets printed as a banner when running the R-view program. The banner is:
+
+# R-view :: R-view - basepairs(RNA) and contacts(RNA/peptides) from a pdb file
+# R-view 0.1 (August 2018)
+# Copyright (C) 2018 Howard Hughes Medical Institute.
+# Freely distributed under the GNU General Public License (GPLv3).
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+\""" > R-view-banner
+
+# Note: license for FastTree only found in source code
+head -n 42 lib/FastTree/src/FastTree.c > LICENSE-FastTree
+
+# Note: no license for rnaview given in tarball
+echo \"""
+No explicit license in the lib/RNAVIEW directory.
+
+In the Gentoo package repositories it is listed as public-domain:
+
+https://packages.gentoo.org/packages/sci-biology/rnaview
+\""" > LICENSE-rnaview
+
+install_license LICENSE
+install_license LICENSE-Rscape-note-GPLv3
+install_license r2r-license-text.txt
+install_license COPYING-nlopt-direct
+install_license COPYING-nlopt
+install_license COPYING-libdivsufsort
+install_license LICENSE-easel
+install_license LICENSE-hmmer
+install_license LICENSE-infernal
+install_license R-view-banner
+install_license LICENSE-FastTree
+install_license LICENSE-rnaview
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# Notes:
+# - configure demands either sse (i686/x86_64) or vmx (powerpc) vector
+#   instructions (this comes from HMMER)
+# - windows compile fails (see HMMER build, but here also in the lib/R2R subdir)
+# - compile fails on powerpc64le:
+#   "HMMER3 Altivec/VMX only supports bigendian platforms: e.g. ppc64 not ppc64le"
+platforms = supported_platforms(; exclude = p -> Sys.iswindows(p) || proc_family(p) != "intel")
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("appcov", :appcov),
+    ExecutableProduct("esl-afetch", :esl_afetch),
+    ExecutableProduct("esl-reformat", :esl_reformat),
+    ExecutableProduct("FastTree", :FastTree),
+    ExecutableProduct("msafilter", :msafilter),
+    ExecutableProduct("r2r", :r2r),
+    ExecutableProduct("rnaview", :rnaview),
+    ExecutableProduct("R-scape", :R_scape),
+    ExecutableProduct("R-scape-sim", :R_scape_sim),
+    ExecutableProduct("R-scape-sim-nobps", :R_scape_sim_nobps),
+    ExecutableProduct("R-view", :R_view),
+    # perl scripts
+    ExecutableProduct("MetamakeDemos.pl", :MetamakeDemos_pl),
+    ExecutableProduct("pdb_parse.pl", :pdb_parse_pl),
+    ExecutableProduct("r2r_msa_comply.pl", :r2r_msa_comply_pl),
+    ExecutableProduct("SelectSubFamilyFromStockholm.pl", :SelectSubFamilyFromStockholm_pl),
```

Comment:
Note that usually we use `ExecutableProduct` for binary executables (ELF/COFF/MachO files), not interpreted scripts, because they aren't standalone (modulo required libraries) and need an interpreter to actually run.  `FileProduct` should be used instead.

Reply:
I made the change, and also removed the `MetamakeDemos.pl` script as it needed the source tarball directory to work.

I guess there could be some difficulty in using the `FileProduct` scripts in the case that they in turn call `ExecutableProduct`s, as one would have to manually set up the environment (`setenv`) like it is done for an `ExecutableProduct`. I eyeballed that this is not the case here.

Just for future reference, if that were to be the case (script calling executable installed by _jll), would one then use `ExecutableProduct` or still choose `FileProduct`?

Reply:
> Just for future reference, if that were to be the case (script calling executable installed by _jll), would one then use ExecutableProduct or still choose FileProduct?

Eh, good question.  Maybe we could do an exception for those cases, but it remains the problem that you may not be able to run the script if you're missing the interpreter at the expected path (Windows being the usual suspect)